### PR TITLE
3718 – Update how we retry tests

### PR DESCRIPTION
**Pull request by vasconsaurus**
## Description

When running locally we might not want the tests to retry, to do this
we would need to comment out the retry line. We want to make this easier
on the developer

Converting the env variable to an integer because ENV[] on ruby returns a string
https://ruby-doc.org/core-3.0.0/ENV.html

Set TEST_RETRY_COUNT to 3 on Travis because we were using the default
before (which is 3)
https://www.rubydoc.info/gems/minitest-retry/

References: 3718

## How has this been tested?

Changed a test to make it fail.
Ran tests locally without setting TEST_RETRY_COUNT, test failed and didn't retry.
Set the env var inside the container (export TEST_RETRY_COUNT=3) and ran the tests again, it retried 3 times.

### CODE REVIEW

**Comment by vasconsaurus** [comment]
Got two errors here, but they seem unrelated to this

```
# functional-tests

Failure:
GraphqlController3Test#test_should_set_smooch_user_slack_channel_url_in_background 
rails test test/controllers/graphql_controller_3_test.rb:344
```

```
# units-test

Error:
SecurityMailerTest#test_should_send_security_notification:
rails test test/mailers/security_mailer_test.rb:4
```

File: .travis.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ before_script:
 - docker-compose -f docker-compose.yml -f docker-test.yml up -d
 - tail -f log/test.log &
 - until curl --silent -I -f --fail http://localhost:3000 ; do printf .; sleep 1; done
-- docker-compose exec api test/setup-parallel-env.sh
+- docker-compose exec -e TEST_RETRY_COUNT=3 api test/setup-parallel-env.sh
```
Comment (caiosba) [inline .travis.yml:24]: @vasconsaurus , we should pass that environment variable in these `script` lines here, no?

https://github.com/meedan/check-api/blob/develop/.travis.yml#L47-L53

Comment (vasconsaurus) [reply .travis.yml:24]: yes! missed those, will update

**Review by caiosba — APPROVED** [review]
Good to go once tests pass.

**Comment by vasconsaurus** [comment]
@caiosba we still have one failing test (in functional-tests)
```
Failure:
GraphqlController3Test#test_should_set_smooch_user_slack_channel_url_in_background 
Expected: "http://SLKIZZDJXE.com"
  Actual: nil
rails test test/controllers/graphql_controller_3_test.rb:344
```

WDYT?

**Comment by caiosba** [comment]
> @caiosba we still have one failing test (in functional-tests)
> 
> ```
> Failure:
> GraphqlController3Test#test_should_set_smooch_user_slack_channel_url_in_background 
> Expected: "http://SLKIZZDJXE.com"
>   Actual: nil
> rails test test/controllers/graphql_controller_3_test.rb:344
> ```
> 
> WDYT?

It's a flaky one. You can merge if it's the only failure.

**Pull request merged**